tests/test_helpers.py

```diff
@@ -515,6 +516,106 @@ def test_proxies_from_env_http_with_auth(mocker) -> None:
     assert proxy_auth.encoding == "latin1"
 
 
+# --------------------- get_env_proxy_for_url ------------------------------
+
+
+@pytest.mark.parametrize(
+    ("proxies", "url_input", "expected_err_msg"),
```

Comment:
Since the `proxies` arg is used for env vars, it's probably best to rename it to reflect this with `proxies_env`, or `proxy_env_vars`, or something similar. The intent is to communicate what this var is better without forcing the readers to try understanding the test implementation.

Reply:
Changing to `proxy_env_vars`, I like that.